app/lib/publisher.rb

```diff
@@ -0,0 +1,68 @@
+require "services"
+
+class Publisher
+  def self.publish_all(types: nil)
+    new(types).publish_all
+  end
+
+  def publish_all
+    types.each do |document_type|
+      publish_document_type(document_type)
+    end
+  end
+
+private
+
+  attr_reader :types
+
+  def initialize(types)
+    @types = types.presence || document_types
+  end
+
+  def publish_document_type(document_type)
+    content_id_and_locale_pairs_for_document_type(
+      document_type,
+    ).each do |content_id, locale|
+      document = Document.find(content_id, locale)
+
+      unless Services.with_timeout(30) { document.publish }
+        Rails.logger.warn "Cannot publish document: #{content_id}:#{locale}"
+      end
+    end
+  end
+
+  def content_id_and_locale_pairs_for_document_type(document_type)
+    unless document_types.include?(document_type)
+      raise ArgumentError, "Unknown document_type: '#{document_type}'"
+    end
+
+    with_timeout(30) do
+      Services.publishing_api.get_content_items(
+        document_type: document_type,
+        publication_state: "draft",
+        fields: %i[content_id locale],
+        per_page: 999_999,
+        order: "updated_at",
+      )["results"].map { |r| r.values_at("content_id", "locale") }
+    end
+  end
+
+  def document_types
+    @document_types ||= all_document_types
+  end
+
+  def all_document_types
+    Rails.application.eager_load!
+    Document.subclasses.map(&:document_type)
+  end
+
+  def with_timeout(seconds)
```

Comment:
I think we can use `Services.with_timeout` here now.

Reply:
Yes! And I did that change. I must have messed around too much with git again :( will fix it shortly

Reply:
Ah no git mess. I just didn't get rid of this implementation :) should be fine now. Will rebase again